colossalai/communication/utils.py

```diff
@@ -107,3 +108,43 @@ def gather_split_1d_tensor(tensor):
     chunks = [gathered[i*numel:(i+1)*numel] for i in range(world_size)]
     dist.all_gather(chunks, tensor, group=gpc.get_group(ParallelMode.PARALLEL_1D))
     return gathered
+
+
+def check_single_machine_multi_gpu_p2p_available(gpu_list):
+    """Opposite of above function, gather values from model parallel ranks.
+
+    :param gpu_list: Queue of gpu to be tested
+    :type list
+
+    :return True or False: gpus returns True if both can access each other via peer to peer, otherwise returns False
+    :rtype bool
+    """
+    logging.basicConfig(format='%(asctime)s - %(pathname)s[line:%(lineno)d] - %(levelname)s: %(message)s', level = logging.DEBUG)
+    unavailble = []
+    check_p2p_result = subprocess.Popen("./simpleP2P", shell=False, stdout=subprocess.PIPE)
```

Comment:
I do not think it's a good idea to pack a 700KB binary file inside the source code.

Reply:
python may not be able to check if peer to peer is available.